kong/dao/factory.lua

```diff
@@ -128,8 +133,50 @@ function _M.new(kong_config)
   return setmetatable(self, _M)
 end
 
+function _M:check_version_compat(min, deprecated)
+  local db_infos = self:infos()
+  if db_infos.version == "unknown" then
+    return nil, "could not check database compatibility: version " ..
+                "is unknown (did you call ':init'?)"
+  end
+
+  local db_v = version.version(db_infos.version)
+  local min_v = version.version(min)
```

Comment:
you can update the 'version.lua' library to 1.0 and change those into `version(min)` without the double 'version' name

Reply:
Yes, I saw; but as usual when possible, we'll make that a separate contribution